### PR TITLE
Upgrading parser library, adding support for CURRENT_DATE and NOT LIKE

### LINKED ISSUE #82: Support CURRENT_DATE()
Please add support for `CURRENT_DATE()`.

I currently get `Error: Function:CURRENT_DATE not available`.

My query:

```sql
SELECT COUNT(*) as count, MONTH(data.returnDate) as month 
FROM package 
WHERE YEAR(data.returnDate) = YEAR(CURRENT_DATE()) 
GROUP BY MONTH(data.returnDate)
```

### LINKED ISSUE #83: NOT LIKE
Is there a way to achieve a NOT LIKE statement?  Thanks!

### PR DESCRIPTION
**Pull request by eXigentCoder**
The `node-sql-parser` library was quite a bit behind and contained a bunch of fixes for issues in parsing sql statements. Most notably with postgresql column and table names wrapped with either a double or single quote. Upgrading this allows us to fix a bunch of downstream bugs that were caused by incorrect parsing.

The library now supports NOT LIKE statements in both the query and where clause.
The library now allows you to use the CURRENT_DATE() function in either the where or query clause.

- closes #82
- closes #83

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,33 +53,35 @@
   "dependencies": {
     "@synatic/json-magic": "^1.0.0",
     "@synatic/type-magic": "^1.0.0",
-    "bson-objectid": "2.0.3",
-    "check-types": "11.1.2",
+    "@types/mocha": "^10.0.1",
```
Comment (thiren) [inline package.json:56]: this should probably be in dev dependencies

File: package.json [hunk 2]
```diff
@@ -52,33 +53,35 @@
...
+    "bson-objectid": "^2.0.4",
```
Comment (thiren) [inline package.json:57]:
```suggestion
    "bson-objectid": "2.0.4",
```

File: package.json [hunk 3]
```diff
@@ -52,33 +53,35 @@
...
+    "check-types": "^11.2.2",
```
Comment (thiren) [inline package.json:58]:
```suggestion
    "check-types": "11.2.2",
```

File: package.json [hunk 4]
```diff
@@ -52,33 +53,35 @@
...
     "deepmerge": "4.2.2",
-    "node-sql-parser": "4.0.2"
+    "node-sql-parser": "^4.6.4"
```
Comment (thiren) [inline package.json:60]:
```suggestion
    "node-sql-parser": "4.6.4"
```

**Review by thiren — CHANGES_REQUESTED** [review]

**Review by filepounder — DISMISSED** [review]

**Review by thiren — APPROVED** [review]

**Pull request merged**